mempool/src/shared_mempool/tasks.rs

```diff
@@ -55,7 +55,26 @@ pub(crate) fn execute_broadcast<V>(
     let network_interface = &smp.network_interface.clone();
     // If there's no connection, don't bother to broadcast
     if network_interface.app_data().read(&peer).is_some() {
-        network_interface.execute_broadcast(peer, backoff, smp);
+        if let Err(err) = network_interface
+            .execute_broadcast(peer, backoff, smp)
+            .await
+        {
+            match err {
+                BroadcastError::NetworkError(peer, error) => error!(LogSchema::event_log(
+                    LogEntry::BroadcastTransaction,
+                    LogEvent::NetworkSendFail
+                )
+                .peer(&peer)
+                .error(&error)),
+                BroadcastError::TooManyPendingBroadcasts(_)
```

Comment:
Can we not replace this with a default match guard? It looks like they're all covered, right?

Reply:
We can.  I followed up with this in a commit that's not on here, I'll backport it.